include/sys/asm.h

```diff
@@ -22,49 +22,84 @@ modified versions thereof.
 extern "C" {
 #endif
 
-#define _MIPS_ISA_MIPS1	1	/* R2/3K */
-#define _MIPS_ISA_MIPS2	2	/* R4K/6K */
-#define _MIPS_ISA_MIPS3	3	/* R4K */
-#define _MIPS_ISA_MIPS4	4	/* TFP */
+#define _MIPS_ISA_MIPS1 1   /* R2/3K */
+#define _MIPS_ISA_MIPS2 2   /* R4K/6K */
+#define _MIPS_ISA_MIPS3 3   /* R4K */
+#define _MIPS_ISA_MIPS4 4   /* TFP */
 
-#define _MIPS_SIM_ABI32		1	/* MIPS MSIG calling convention */
-#define _MIPS_SIM_NABI32	2	/* MIPS new 32-bit abi */
-		/* NABI32 is 64bit calling convention but 32bit type sizes) */
-#define _MIPS_SIM_ABI64		3	/* MIPS 64 calling convention */
+#define _MIPS_SIM_ABI32     1   /* MIPS MSIG calling convention */
+#define _MIPS_SIM_NABI32    2   /* MIPS new 32-bit abi */
+        /* NABI32 is 64bit calling convention but 32bit type sizes) */
+#define _MIPS_SIM_ABI64     3   /* MIPS 64 calling convention */
 
-#define	LEAF(x)						\
-	.globl	x;					\
-	.ent	x,0;					\
-x:;							\
-	.frame	sp,0,ra
 
-#define	XLEAF(x)					\
-	.global x;
+#define LEAF(x)             \
+    .globl  x              ;\
+    .align  2              ;\
+    .ent    x,0            ;\
+    x:                     ;\
+    .frame  sp,0,ra
+    //.type   x, @function   ;\
 
-#define	END(proc)					\
-	.end	proc
+#if defined(BBPLAYER) || defined(__sgi)
+#define XLEAF(x)    \
+    .globl  x      ;\
+    x:
+#else
+#define XLEAF(x)    \
+    .globl  x
+#endif
+
+#ifdef BBPLAYER
+#define END(proc)           \
+    .end    proc           ;\
+    .size   proc, . - proc
+#else
+#define END(proc)           \
+    .end    proc
+#endif
+
+#define ABS(x, y)   \
+    .globl  x      ;\
+    x   =   y
 
-#define	ABS(x, y)					\
-	.globl	x;					\
-	x	=	y
-	
-#define	EXPORT(x)					\
-	.globl	x;					\
-x:
+#define EXPORT(x)   \
+    .globl  x      ;\
+    x:
 
 #ifdef BBPLAYER
```

Comment:
```suggestion
#if defined(BBPLAYER) || defined(__sgi)
```
If you want, this looks to be exactly what irix uses.